read_receipts: Disable showing read receipts for Notification Bot messages.

Riken-Shah opened the pull request:
The implementation is simple, we just check if the message sender is a notification bot to decide if we should show the read receipts list.

Fixes #22905



<!-- Describe your pull request here.-->



<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<img width="450" alt="Screenshot 2022-09-23 at 12 54 00 PM" src="https://user-images.githubusercontent.com/63820270/191910554-2251dbbb-07fa-4fb9-b5b3-e6d2d7b0dbf7.png">




**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

Riken-Shah commented:
Instead of only applying the disable logic for the notification bot I have applied the logic for all the bots.

@alya Is this fine?

Also updated suggested text from, 
> Read receipts are not available for ~Notification~ any Bot messages.

alya commented:
Hm, could we just apply the logic to Notification Bot? Other bots don't send messages that the app automatically marks as read.

Riken-Shah commented:
> Hm, could we just apply the logic to Notification Bot? Other bots don't send messages that the app automatically marks as read.

Cool Done!

juliaBichler01 reviewed (commented):
The changes to the code look good to me, but one question/suggestion about how the popup window looks: 
Is it on purpose that there is a lot of empty space below the text? I think it would look better if this would be a bit less.

Riken-Shah commented:
Thanks for the review!

> Is it on purpose that there is a lot of empty space below the text? I think it would look better if this would be a bit less.

We show the simar space if there are no read receipts.

<img width="406" alt="Screenshot 2022-09-11 at 12 40 09 AM" src="https://user-images.githubusercontent.com/63820270/189498348-8919ee67-ba59-46ac-9068-ffa65071bebe.png">


I also agree it'd look better if we remove extra space. @alya What do you think?

timabbott commented:
The design challenge with the popover window is that we show it while the loading indicator is pinning, with a certain height, and then its height potentially changes when the data contents are added; it feels not great. So I think we have a minimum height set in the CSS for what will look good if there's at least one user shown, for example, read receipts for a private message.

One possible solution would be to just not have the modal have a loading indicator -- just only open it once we've received a response from the server (if we're making one); that way, you never see the thing's size change shortly after it opens.

The downside of that approach would be that you get no feedback that your click was processed properly..

Riken-Shah commented:
We don't make any API call for notification bot messages, so for this case, we can remove the minimum height rule.

Riken-Shah commented:
>  We don't make any API call for notification bot messages, so for this case, we can remove the minimum height rule.



@alya Should I implement this?

alya commented:
> > We don't make any API call for notification bot messages, so for this case, we can remove the minimum height rule.
> 
> @alya Should I implement this?

Yes, that sounds good to me!

Riken-Shah commented:
@alya & @timabbott This is ready for another round of review!

alya commented:
Hmm, I think we went too far with reducing the height required -- the modal looks cut off on the bottom.

Can we try leaving as much space on the bottom as we do on the top?

alya commented:
I didn't see any other issues in manual testing.

Riken-Shah commented:
> Can we try leaving as much space on the bottom as we do on the top?

On top we have `10px` margin, at bottom we had `8px` padding I have updated it to `10px`. 
Let me know if it looks alright or if I need to add more spacing. 

P.S I have updated the screenshot on the top comment.

alya commented:
Hmm, it seems like there must be additional margin coming from somewhere -- I'm still seeing more space on top than on the bottom.

alya commented:
@timabbott do you want to take a look?

Riken-Shah commented:
>  I'm still seeing more space on top than on the bottom.

Hmm, maybe I can add one more `px` to compensate for `hr` border, but not sure that will create any significant difference.

timabbott commented:
I think Alya had a different existing padding as what to match -- there's 16px of padding above the heading, and 10px below; so we should just do the 16px, not 10px. It lgtm changing from 10px to 16px:

![image](https://user-images.githubusercontent.com/2746074/192068562-0a560023-ced0-4684-87cb-f18ab2f521db.png)


Merged with that tweak, thanks @Riken-Shah!

timabbott commented:
Merged, with a bonus commit to move the read receipts CSS next to the related modal CSS.
